packages/webpack/src/plugins/plugin.ts

```diff
@@ -0,0 +1,185 @@
+  Record<string, TargetConfiguration>
+> = {};
+
+function readTargetsCache(): Record<
+  string,
+  Record<string, TargetConfiguration>
+> {
+  return readJsonFile(cachePath);
+}
+
+function writeTargetsToCache(
+  targets: Record<string, Record<string, TargetConfiguration>>
+) {
+  writeJsonFile(cachePath, targets);
+}
+
+export const createDependencies: CreateDependencies = () => {
+  writeTargetsToCache(calculatedTargets);
+  return [];
+};
+
+export const createNodes: CreateNodes<WebpackPluginOptions> = [
+  '**/webpack.config.{js,ts,mjs,mts,cjs,cts}',
+  async (configFilePath, options, context) => {
+    options ??= {};
+    options.buildTargetName ??= 'build';
+    options.serveTargetName ??= 'serve';
+    options.staticServeTargetName ??= 'static-serve';
+    options.previewTargetName ??= 'preview';
+
+    const projectRoot = dirname(configFilePath);
+
+    // Do not create a project if package.json and project.json isn't there.
+    const siblingFiles = readdirSync(join(context.workspaceRoot, projectRoot));
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    const hash = calculateHashForCreateNodes(projectRoot, options, context, [
+      getLockFileName(detectPackageManager(context.workspaceRoot)),
+    ]);
+    const targets = targetsCache[hash]
+      ? targetsCache[hash]
+      : await createWebpackTargets(
+          configFilePath,
+          projectRoot,
+          options,
+          context
+        );
+
+    return {
+      projects: {
+        [projectRoot]: {
+          projectType: 'application',
+          targets,
+        },
+      },
+    };
+  },
+];
+
+async function createWebpackTargets(
+  configFilePath: string,
+  projectRoot: string,
+  options: WebpackPluginOptions,
+  context: CreateNodesContext
+): Promise<
+  Record<
+    string,
+    TargetConfiguration<WebpackExecutorOptions | WebDevServerOptions>
+  >
+> {
+  const namedInputs = getNamedInputs(projectRoot, context);
+  const webpackConfig = resolveUserDefinedWebpackConfig(
+    join(context.workspaceRoot, configFilePath),
+    getRootTsConfigPath()
+  );
+  const webpackOptions = await readWebpackOptions(webpackConfig);
+
+  const outputPath =
+    normalizeOutputPath(webpackOptions.output?.path) ??
+    '{workspaceRoot}/dist/{projectRoot}';
+
+  const targets = {};
+
+  const configBasename = basename(configFilePath);
+
+  targets[options.buildTargetName] = {
+    command: `webpack -c ${configBasename} --node-env=production`,
+    options: {
+      cwd: projectRoot,
+    },
+  };
+
+  const buildTargetDefaults = readTargetDefaultsForTarget(
+    options.buildTargetName,
+    context.nxJsonConfiguration.targetDefaults
+  );
+
+  if (buildTargetDefaults?.cache === undefined) {
+    targets[options.buildTargetName].cache = true;
+  }
+
+  if (buildTargetDefaults?.inputs === undefined) {
+    targets[options.buildTargetName].inputs =
+      'production' in namedInputs
```

Comment:
This should now have an input of the `webpack-cli`.

https://github.com/nrwl/nx/blob/a57122d47c0b9d2ae94facd5a865645093067342/packages/cypress/src/plugins/plugin.ts#L338C41-L338C41